### ISSUE TITLE
Directory Issue

### ISSUE THREAD

**Issue by rafaeltp** [masc-ucsc/cmpe220fall16#4]
The Directory unit test is broken. The issue seems to be coming from the signal ```drtomem_pfreq```, which is assigned to from line 89, and then again by the fflop on line 96.

```  assign drtomem_pfreq.paddr = drff_pfreq.paddr;```

```
  fflop #(.Size(50)) ff0 (
   //...
  .q        (drtomem_pfreq),
   //...
```

There are two ways of solving this in a cleaner way:

1) changing the signal passed to the port din of the fflop
2) creating a signal that takes all the fileds of drtomem_pfreq, except paddr, and takes paddr from drff_pfreq

I was not able to get a consistent message from verilator, but the most accurate error message seems to be:

```
%Error-BLKANDNBLK: tests/directory_bank_wp.v:89: Unsupported: Blocked and non-blocking assignments to same variable: drtomem_pfreq_paddr
%Error: Exiting due to 1 error(s)
%Error: See the manual and http://www.veripool.org/verilator for more assistance.
%Error: Command Failed /usr/local/bin/verilator_bin --assert --debug-check -I./rtl --Wall --cc --trace --top-module directory_bank_wp ./tests/directory_bank_wp.v ./rtl/directory_bank.v ./rtl/fflop.v ./rtl/flop_r.v --exe tests/directory_bank_wp_tb.cpp -CFLAGS '-DTRACE=1'
make: *** [Makefile:58: directory_bank_wp] Error 10
```

Edit: fixed style

**Comment by hznichol** [comment]
This was a typo. I fixed it, but my test is still failing. I believe my testbench is generating an input that is causing the fflop to fail somehow. Still trying to figure it out, so it might be a little while. I included a screenshot of the waveform of the fflop and the test fails at the end.
![fflop_waveform_fail](https://cloud.githubusercontent.com/assets/8183930/19670197/2cc845a6-9a1a-11e6-9c72-0d7571dfb32c.PNG)
